src/argilla_sdk/datasets/_export.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+import json
+import logging
+import os
+from pathlib import Path
+import warnings
+from typing import Optional, Union, TYPE_CHECKING, Tuple
+from uuid import uuid4
+
+from argilla_sdk._models import DatasetModel
+from argilla_sdk.client import Argilla
+from argilla_sdk.settings import Settings
+from argilla_sdk.workspaces._resource import Workspace
+
+
+if TYPE_CHECKING:
+    from argilla_sdk import Dataset
+    from argilla_sdk.records import DatasetRecords
+
+
+class DiskImportExportMixin(ABC):
+    """A mixin for exporting and importing datasets to and from disk."""
+
+    _model: DatasetModel
+    _default_settings_path = "settings.json"
+    _default_records_path = "records.json"
+    _default_dataset_path = "dataset.json"
+
+    def to_disk(self, path: str) -> str:
```

Comment:
If we include a type for self, we don't need to define abstract methods.
```suggestion
    def to_disk(self: "Dataset", path: str) -> str:
```